packages/protocol/contracts/L1/IProver.sol

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: MIT
+//  _____     _ _         _         _
+// |_   _|_ _(_) |_____  | |   __ _| |__ ___
+//   | |/ _` | | / / _ \ | |__/ _` | '_ (_-<
+//   |_|\__,_|_|_\_\___/ |____\__,_|_.__/__/
+
+pragma solidity ^0.8.20;
+
+import { TaikoData } from "./TaikoData.sol";
+
+/// @title IProver Interface
+/// @notice Defines the function that handle prover assignment.
+interface IProver {
```

Comment:
I try not to add any "notification" call to IProver during verification to avoid the transactor to spend gas on such calls. If the prover assignment can be identified uniquely, the prover should be able to do the book keeping offchain.

Reply:
> Add the blockID so there can be a unique mapping between a block and an internal prover
Done

> onBlockSlashed
This is not added, lets see if it is necessary after the next testnet